### ISSUE TITLE
Improve Clickable and Input Component

### ISSUE THREAD

**Issue by SirFurness** [revery-ui/revery#857]
Some ideas for improving the functionality of the Clickable and Input component:

- Clicking inside of an input should change the cursor position on mouse down, not mouse up.

- The cursor should still be captured by the input when it leaves the input so that highlighting text can work well.

For Clickable:

- Clickable should use `Hooks.mouseCapture` instead of its own mouse capture system

- Mouse down on a clickable, leaving the clickable, re-entering the clickable, and then mouse up back on the clickable should still count as a click. 
In other words, if the mouse down and mouse up occur on the clickable, it should not matter if the cursor left.
